### PR TITLE
feat(typescript): update to 1.9 nightly.

### PR DESCRIPTION
**Pull request by alexeagle**
Tested by build.js.cjs, copy to another folder, and manually run compiler to cause 1.8 to type-check the .d.ts files.
It would be better to have a test that runs ts1.8 typechecker in travis, but it's a real pain to change versions of a dependency partway through a build.

### CODE REVIEW

**Comment by alexeagle** [comment]
note, requires #8002

File: tools/broccoli/trees/node_tree.ts [hunk 1]
```diff
@@ -120,6 +120,15 @@ module.exports = function makeNodeTree(projects, destinationPath) {
 
   let compiledTree = mergeTrees([compiledSrcTree, compiledTestTree]);
 
+  // Down-level .d.ts files to be TS 1.8 compatible
+  compiledTree = replace(compiledTree, {
+    files: ['**/*.d.ts'],
+    patterns: [
+      {match: /^(\s*)readonly\s+/mg, replacement: "$1"},
```
Comment (jeffbcross) [inline tools/broccoli/trees/node_tree.ts]: What is this supposed to match?

**Comment by jeffbcross** [comment]
> It would be better to have a test that runs ts1.8 typechecker in travis, but it's a real pain to change versions of a dependency partway through a build.

Would it be significant work (or an otherwise bad idea) to add a simple Travis job that installs TS 1.8 and runs `test.unit.cjs`, then remove typescript from node_modules so it doesn't get cached?

File: tools/broccoli/trees/node_tree.ts (hunk 1 above)
Comment (alexeagle) [inline tools/broccoli/trees/node_tree.ts]: matches 
   readonly foo: any;
replace with
   foo: any;

are you asking for a comment here to explain?

Comment (jeffbcross) [inline tools/broccoli/trees/node_tree.ts]: Yeah, I didn't understand the purpose. A code comment with a link would be helpful so we can confidently remove this someday when 1.8 is no longer supported.

Comment (alexeagle) [inline tools/broccoli/trees/node_tree.ts]: added
` // TODO(alexeagle): this can be removed once we drop support for using Angular 2 with TS 1.8`

**Comment by alexeagle** [comment]
Added a line in `build_and_test.sh` to cover the scenario where we make changes in angular2 that break ts1.8 users.

I don't want to add another matrix entry if possible, since that makes each build consume more travis workers. Since we already have this special typescript_next worker, it now has essentially this logic:
- install typescript@next
- run `gulp build.js`
- install typescript@1.8.9
- run `gulp test.typings`
the latter now attempts to compile all our examples against the locally-built angular.

**Pull request closed without merging**